src/daft-hive/src/lib.rs

```diff
@@ -0,0 +1,96 @@
+use std::collections::HashMap;
+
+use common_error::DaftResult;
+use daft_core::{datatypes::Utf8Array, series::IntoSeries};
+use daft_decoding::inference::infer;
+use daft_schema::{dtype::DaftDataType, field::Field, schema::Schema};
+use daft_table::Table;
+
+/// Parses hive-style /key=value/ components from a uri.
+pub fn parse_hive_partitioning(uri: &str) -> HashMap<&str, &str> {
+    let mut equality_pos = 0;
+    let mut partition_start = 0;
+    let mut valid_partition = true;
+    let mut partitions = HashMap::new();
+    // Loops through the uri looking for valid partitions. Although we consume partitions only when
+    // encountering a slash separator, we never need to grab a partition key-value pair from the end
+    // of the uri because uri's are expected to end in either a filename, or in GET parameters.
```

Comment:
The way I way thinking about this, it's not unusual to have GET parameters in url paths. For example, with S3 we might get pre-signed strings. Not 100% on this, but maybe S3 filtering also works via GET strings.

However it's kinda moot because our glob reader doesn't work well with GET parameters (we only glob after the GET, so `s3://path/to/glob/**?param=val` doesn't glob properly. I prefer to keep it because we might add support in the future